Add docs for mode changes (#311)
saucelabs/sauce-docs: docs/testrunner-toolkit/configuration/common-syntax.md

⚠️ `registry` configuration is only supported in latest cypress docker image. No other frameworks, or Sauce Labs cloud.

## `defaults`

> __Description__: Field for defining default settings.

Reviewer: ```suggestion
__Description__: Field for defining default settings for the project.
```

__Type__: *object*

__Example__:
```yaml
defaults:
  - mode: "sauce"
```

### `mode`

> __Description__: Specifies the default environment in which the tests should run. Choice: docker|sauce. If there is no setting for a suite, it should use this default setting.

Reviewer: ```suggestion
__Description__: Specifies whether tests in the project will run on `docker` or `sauce`. This default setting can be overridden by the `mode` value for individual suites or at runtime by the `--mode` command line parameter.
```

> __Description__: Specifies the default environment in which the tests should run. Choice: docker|sauce. If there is no setting for a suite, it should use this default setting.

Reviewer: Also, I think the description should include the fact that this value is overridden when `mode` is set as a command line parameter.

> __Description__: Specifies whether tests in the project will run on `docker` or `sauce`. This default setting can be overridden by the `mode` value for individual suites or at runtime by the `--test-env` command line parameter.

Reviewer: ```suggestion
__Description__: Specifies whether tests in the project will run on `docker` or `sauce`. This default setting can be overridden by the `mode` value for individual suites.
```

    screenResolution: "1920x1080"
```

### `mode`

> __Description__: Specifies the default environment in which the test suite should run. Choice: docker|sauce. If there is no setting for a suite, it should use this default setting.

Reviewer: ```suggestion
__Description__: Allows you to specify whether the individual suite will run on `docker` or `sauce`, potentially overriding the default project mode setting. The suite will always run in the specified mode unless overridden at runtime by the `--mode` command line parameter.
```

> __Description__: Allows you to specify whether the individual suite will run on `docker` or `sauce`, potentially overriding the default project mode setting. The suite will always run in the specified mode unless overridden at runtime by the `--test-env` command line parameter.

Reviewer: ```suggestion
__Description__: Allows you to specify whether the individual suite will run on `docker` or `sauce`, potentially overriding the default project mode setting.
```
